src/mod/endpoints/mod_sofia/sofia_glue.c

```diff
@@ -1086,6 +1086,13 @@ switch_status_t sofia_glue_do_invite(switch_core_session_t *session)
 	if (switch_channel_test_flag(tech_pvt->channel, CF_RECOVERING)) {
 		switch_telnyx_on_channel_recover(tech_pvt->channel);
 	}
+	
+	/* TEL-5560: If this is set true, it means we previously received a reInvite
+	 * on an outbound leg.  Need to set this to false because we're now going to
+	 * send an outbound invite elsewhere.  (This function we're in gets called up
+	 * in initial invites and cases of 3XX in-dialog)
+	 */
+	switch_channel_set_variable(tech_pvt->channel, "dlg_req_swap_direction", "false");
 
```

Comment:
But, shouldn't this be only set when we recovering the call. The swap is still valid if we have to send a reINVITE on the same host. Except this function isn't called when we send a reINVITE

Reply:
dlg_req_swap_direction controls the swapping of the from & to during recovery (in sofia_glue.c's sofia_recover_callback() function), and thus the issue isn't specific to mod_call_recovery.  The situation would be like this:

1.  b-leg outbound to a SIP UAS elsewhere.
2.  After completion of the first INVITE transaction, one or more re-invites comes back to this FreeSWITCH instance, thus setting dlg_req_swap_direction to true, regardless of what value would be set for sip_invite_transaction_type .  Now the to & from will be swapped due to the direction of the invite inbound.
3.  But another event happens that causes this instance of FreeSWITCH to send a reinvite to the remote UAS, which until now would still have left dlg_req_swap_direction as true, which isn't proper, even if it only affects call recovery.  We want it to reflect the proper state of the channel.  So I set it to false, because once again the to and the from will be in the correct direction if the FreeSWITCH is the one that originates the invite/reinvite.

Reply:
There is no other invite sending function that I am finding.  sofia_glue_do_invite() gets called in 3 or 4 total places between mod_sofia.c and sofia.c , and they're not all for initial invite handling.

Reply:
To be extra clear:  dlg_req_swap_direction only ever gets set to true in the vanilla freeswitch source.  I believe this to be in error, as the direction can change multiple times within the same SIP dialog.  thus I added where I believe makes sense to put it back false.